Document how to work with ClangFormat

paramat opened the pull request:
As discussed elsewhere, the ClangFormat check often makes unsuitable formatting requests and misleads contributors into thinking they must satisfy the all the requests, and we do not warn about this.
One example is #10408 , where the author added 3300 lines of changes to a 700 line change PR to satisfy unsuitable requests made by ClangFormat.
This PR explains how to work with ClangFormat and makes it clear that unsuitable ClangFormat requests should not be applied.
The added text is placed at the end of the 'Code' section.

![Screenshot from 2020-10-07 19-18-05](https://user-images.githubusercontent.com/3686677/95371310-eea65f80-08d1-11eb-8f03-7921529f8cd3.png)

nerzhul commented on .github/CONTRIBUTING.md:
```diff
@@ -35,6 +35,16 @@ Contributions are welcome! Here's how you can help:
+
+When you submit a pull request, GitHub automatically runs checks on the Minetest Engine combined with your changes. One of these checks is called 'cpp lint / clang format', which checks code formatting and checks for code style errors.
+
+Because formatting for readability requires human judgement, and because an automated formatting check has limited configurability, this check often fails and ClangFormat often makes unsuitable formatting requests which make code readability worse.
```
s/often/sometimes

rubenwardy replied:
Under the current terrible config, it is often

paramat replied:
Yes, approx 50% is unsuitable requests in my experience.
Even with improved configuration it will still be 'often' as CF cannot be made to ignore code alignment, apparently.
Best to be honest to contributors about this so that they are prepared.

paramat commented:
Updated. New screenshot in first comment.

* I moved the text to the end of the 'Code' section, seems more suitable.
* The text was a too verbose for an introductory document, so i made it more concise.
* I removed the mention of the whitelist, too technical for this document. Instead i encouraged discussing how to progress with a core developer.

v-rob commented:
Looks good.  Some missing commas, but whatever :)

SmallJoker commented:
Looks good. No objections.
